jax/_src/dtypes.py

```diff
@@ -270,23 +270,34 @@ def _dtype_and_weaktype(value):
   """Return a (dtype, weak_type) tuple for the given input."""
   return dtype(value), any(value is typ for typ in _weak_types) or is_weakly_typed(value)
 
-def _type_promotion_lattice():
+def _type_promotion_lattice(jax_numpy_dtype_promotion):
   """
   Return the type promotion lattice in the form of a DAG.
   This DAG maps each type to its immediately higher type on the lattice.
   """
   b1, u1, u2, u4, u8, i1, i2, i4, i8, bf, f2, f4, f8, c4, c8 = _jax_types
   i_, f_, c_ = _weak_types
-  return {
-    b1: [i_],
-    u1: [i2, u2], u2: [i4, u4], u4: [i8, u8], u8: [f_],
-    i_: [u1, i1], i1: [i2], i2: [i4], i4: [i8], i8: [f_],
-    f_: [bf, f2, c_], bf: [f4], f2: [f4], f4: [f8, c4], f8: [c8],
-    c_: [c4], c4: [c8], c8: [],
-  }
-
-def _make_lattice_upper_bounds():
-  lattice = _type_promotion_lattice()
+  if jax_numpy_dtype_promotion == 'standard':
+    return {
+      b1: [i_],
+      u1: [i2, u2], u2: [i4, u4], u4: [i8, u8], u8: [f_],
+      i_: [u1, i1], i1: [i2], i2: [i4], i4: [i8], i8: [f_],
+      f_: [bf, f2, c_], bf: [f4], f2: [f4], f4: [f8, c4], f8: [c8],
+      c_: [c4], c4: [c8], c8: [],
+    }
+  elif jax_numpy_dtype_promotion == 'strict':
+    return {
+      i_: [f_] + _jax_types[1:9],
```

Comment:
I wonder if it'd be more readable if we gave a variable name to `_jax_types[1:9]` and friends, like above if we split up things like

```python
_exact_types = [...]
_float_types = [...]
_complex_types = [...]
_jax_types = [*_exact_types, *_float_types, *_complex_types]
```

WDYT?

Reply:
Good idea - Done!